### PR TITLE
layan-gtk-theme: init at 2021-06-30

### PR DESCRIPTION
**Pull request by VergeDX**
###### Motivation for this change

Layan is a flat Material Design theme for GTK 3, GTK 2 and Gnome-Shell which supports GTK 3 and GTK 2 based desktop environments like Gnome, Budgie, etc.
-- quote from [vinceliuice/Layan-gtk-theme](https://github.com/vinceliuice/Layan-gtk-theme)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130064` at 86cfd3c0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>layan-gtk-theme</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/data/themes/layan-gtk-theme/default.nix:17:3:
```
   |
17 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/data/themes/layan-gtk-theme/default.nix:26:5:
```
   |
26 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/data/themes/layan-gtk-theme/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+}:
```
Comment (musfay) [inline pkgs/data/themes/layan-gtk-theme/default.nix:4]:
```suggestion
, gtk-engine-murrine
}:
```

File: pkgs/data/themes/layan-gtk-theme/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "layan-gtk-theme";
+  version = "2021-06-30";
+
+  src = fetchFromGitHub {
+    owner = "vinceliuice";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-FI8+AJlcPHGOzxN6HUKLtPGLe8JTfTQ9Az9NsvVUK7g=";
+  };
+
```
Comment (musfay) [inline pkgs/data/themes/layan-gtk-theme/default.nix:16]:
```suggestion

  propagatedUserEnvPkgs = [ gtk-engine-murrine ];

```

File: pkgs/data/themes/layan-gtk-theme/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  installPhase = ''
+    patchShebangs .
+    mkdir -p $out/share/themes
+    ./install.sh -d $out/share/themes -n Layan
+  '';
```
Comment (musfay) [inline pkgs/data/themes/layan-gtk-theme/default.nix:21]:
```suggestion
  installPhase = ''
    runHook preInstall
    patchShebangs .
    mkdir -p $out/share/themes
    ./install.sh -d $out/share/themes -n Layan
    runHook postInstall
  '';
```

**Review by musfay — CHANGES_REQUESTED** [review]

**Comment by VergeDX** [comment]
@musfay Changes performed, thanks for reviewing!

**Review by musfay — APPROVED** [review]
LGTM

File: pkgs/data/themes/layan-gtk-theme/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    patchShebangs .
+    mkdir -p $out/share/themes
+    ./install.sh -d $out/share/themes -n Layan
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A flat Material Design theme for GTK 3, GTK 2 and Gnome-Shell.";
+    homepage = "https://github.com/vinceliuice/Layan-gtk-theme";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/data/themes/layan-gtk-theme/default.nix:31]: Do you know if the license is gpl3Only or gpl3Plus?

File: pkgs/data/themes/layan-gtk-theme/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-FI8+AJlcPHGOzxN6HUKLtPGLe8JTfTQ9Az9NsvVUK7g=";
+  };
+
+  propagatedUserEnvPkgs = [ gtk-engine-murrine ];
+
+  installPhase = ''
+    runHook preInstall
+    patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/data/themes/layan-gtk-theme/default.nix:22]: Please move this to postPatch and be more specific with the path.

**Comment by VergeDX** [comment]
The `name` env variable will affect the `install.sh` script, so I unset it before install.
see https://github.com/vinceliuice/Layan-gtk-theme/blob/2021-06-30/install.sh#L189

**Comment by VergeDX** [comment]
Also, gpl3 changes to gpl3Only.

File: pkgs/data/themes/layan-gtk-theme/default.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchFromGitHub {
+    owner = "vinceliuice";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-FI8+AJlcPHGOzxN6HUKLtPGLe8JTfTQ9Az9NsvVUK7g=";
+  };
+
+  propagatedUserEnvPkgs = [ gtk-engine-murrine ];
+
+  postPatch = "patchShebangs install.sh";
```
Comment (SuperSandro2000) [inline pkgs/data/themes/layan-gtk-theme/default.nix:20]:
```suggestion
  postPatch = ''
    patchShebangs install.sh
  '';

```

**Pull request merged**